src/uct/tcp/tcp_ep.c

```diff
@@ -928,9 +925,12 @@ ucs_status_t uct_tcp_ep_am_short(uct_ep_h uct_ep, uint8_t am_id, uint64_t header
             if (uct_tcp_ep_ctx_buf_need_progress(&ep->tx)) {
                 /* Copy only user's header and payload to the TX buffer,
                  * TCP AM header is placed at the beginning of the buffer */
+                offset = ((ep->tx.offset >= sizeof(*hdr)) ?
```

Comment:
maybe if `ep->tx.offset < sizeof(*hdr)`, we should not call ucs_iov_copy at all?